crates/ruff_linter/src/checkers/logical_lines.rs

```diff
@@ -47,34 +47,90 @@ pub(crate) fn check_logical_lines(
 
     for line in &LogicalLines::from_tokens(tokens, locator) {
         if line.flags().contains(TokenFlags::OPERATOR) {
-            space_around_operator(&line, &mut context);
-            whitespace_around_named_parameter_equals(&line, &mut context);
-            missing_whitespace_around_operator(&line, &mut context);
-            missing_whitespace(&line, &mut context);
+            if settings.rules.any_enabled(&[
+                Rule::MultipleSpacesBeforeOperator,
+                Rule::MultipleSpacesAfterOperator,
+                Rule::TabBeforeOperator,
+                Rule::TabAfterOperator,
+            ]) {
+                space_around_operator(&line, &mut context);
+            }
+
+            if settings.rules.any_enabled(&[
+                Rule::UnexpectedSpacesAroundKeywordParameterEquals,
+                Rule::MissingWhitespaceAroundParameterEquals,
+            ]) {
```

Comment:
Let's hope the compiler is smart enough to move the check out of the `for` loop :)

Reply:
Yeah, I hope so as well. Do you know how can one get that kind of information?

Reply:
I've decided to collect them out in variables outside the loop similar to `physical_lines.rs`.